blockchain/blockchain_test.go

```diff
@@ -398,3 +398,26 @@ func TestState(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestL1Update(t *testing.T) {
+	heads := []*core.L1Head{
+		{
+			BlockNumber: 1,
+			StateRoot:   new(felt.Felt).SetUint64(2),
+		},
+		{
+			BlockNumber: 2,
+			StateRoot:   new(felt.Felt).SetUint64(3),
+		},
+	}
+
+	for _, head := range heads {
+		t.Run("update L1 head to block "+string(rune(head.BlockNumber)), func(t *testing.T) {
```

Comment:
I think you wanted to do smth. like `strconv.Itoa(...)`, currently `string(rune(...))` expression returns A letter and some weird rune LTX for block 1 and 2 respectively